core-backend/wasmi/src/funcs/mod.rs

```diff
@@ -430,17 +430,13 @@ where
         at: u32,
         len: u32,
     ) -> Result<&'_ [u8], FuncError<<E as Ext>::Error>> {
-        let msg = ext.read().map_err(FuncError::Core)?;
+        let msg = ext.read(at, len).map_err(FuncError::Core)?;
 
-        let last_idx = at
-            .checked_add(len)
-            .ok_or_else(|| FuncError::ReadLenOverflow(at, len))?;
+        // 'at' and 'len' correct and saturation checked in Ext::read
+        debug_assert!(at.checked_add(len).is_some());
```

Comment:
We have tests only for `--release`, so I think better to use `unreachable!` here

Reply:
unreachable will effect on release build, but this line not